ct-app/economic_handler/economic_handler.py

```diff
@@ -149,6 +210,62 @@ async def get_unique_safe_peerId_links(self):
 
         return "unique_peer_safe_links", response
 
+    async def get_database_metrics(self):
+        """
+        This function establishes a connection to the database using the provided
+        connection details, retrieves the latest peer information from the database
+        table, and returns the data as a dictionary.
+        """
+        with DatabaseConnection(
+            envvar("DB_NAME"),
+            envvar("DB_HOST"),
+            envvar("DB_USER"),
+            envvar("DB_PASSWORD"),
+            envvar("DB_PORT", int),
+        ) as db:
+            try:
+                table_name = "raw_data_table"
+                last_added_rows = db.last_added_rows(table_name)
+
+                metrics_dict = {}
+                for row in last_added_rows:
+                    id, peer_id, netw_ids, latency_metrics, timestamp = row
+
+                    metrics_dict[peer_id] = {
+                        "netw_ids": netw_ids,
+                        "latency_metrics": latency_metrics,
+                        "id": id,
+                        "Timestamp": timestamp,
+                    }
+
+                return "metricsDB", metrics_dict
+
+            except ValueError as e:
+                log.exception(f"Error reading data from database table: {e}")
+            return "metricsDB", {}
```

Comment:
Logging the exception variable `e` not necessary, already included in `log.exception(...)`
```suggestion
            except ValueError:
                log.exception(f"Error reading data from database table")
            return "metricsDB", {}
```

Reply:
Nice catch. I am making this mistake all the time :D I will make a new commit as f string is not needed without any variable.